src/base-implementation.ts

```diff
@@ -0,0 +1,103 @@
+import { Wallet } from '@harmony-js/account'
+import { Contract } from '@harmony-js/contract'
+import { Harmony } from '@harmony-js/core'
+import { BigNumber, logger, constants } from 'ethers'
```

Comment:
If we are going to use @harmony-js sdk, we should just use the individual packages from etherts. like in this case `import { BigNumber } from '@ethersproject/bignumber` or check in the @harmony-js sdk if they have something for handle BigNumbers, logger and other things that we'll need